src/django/api/management/commands/export_csv.py

```diff
@@ -18,42 +19,65 @@ def print_with_time(text: str) -> None:
     print(f"{datetime.now().strftime('%Y-%m-%d %H:%M:%S')} -- {text}")
 
 
-class Command(BaseCommand):
-    help = 'Export all facilities to a CSV file.'
+def create_chunk(list, size, index):
+    return list[index:index + size]
 
-    def add_arguments(self, parser):
-        return
-        print("add_arguments")
 
-    def handle(self, *args, **options):
-        print_with_time("Started indexing facilities")
-        index_facilities_new()
-        facilities = FacilityIndex.objects.order_by("id").all()
-        count = facilities.count()
-        print_with_time(f"Finished indexing {count} facilities")
+def create_with_headers():
+    now = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')
+    filename = f"./facilities-command-{now}.csv"
+    with open(filename, 'w+') as f:
+        writer = csv.writer(f)
+        print_with_time("Created file")
 
-        serializer = FacilityIndexDownloadSerializer()
+        headers = serializer.get_headers()
+        writer.writerow(headers)
+        print_with_time("Written headers")
 
-        now = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')
-        with open(f"./facilities-command-{now}.csv", 'w+') as f:
-            writer = csv.writer(f)
-            print_with_time("Opened file")
+    return filename
 
-            writer.writerow(serializer.get_headers())
-            print_with_time("Written headers")
 
-            print_with_time("Fetching data from DB")
-            i = 0
-            for facility in facilities:
-                if i == 0:
-                    print_with_time(f"Started exporting {count} facilities")
+def modify(filename, chunk):
+    with open(filename, 'a') as f:
+        writer = csv.writer(f)
+        for facility in chunk:
+            row = serializer.get_row(facility)
+            writer.writerow(row)
 
-                i += 1
 
-                row = serializer.get_row(facility)
-                writer.writerow(row)
+class Command(BaseCommand):
+    help = 'Export all facilities to a CSV file.'
 
-                if i % 100 == 0:
-                    print_with_time(f"Wrote {count_with_percent(i, count)}")
+    def add_arguments(self, parser):
+        return
 
-            print_with_time(f"Finished {count_with_percent(i, count)}")
+    def handle(self, *args, **options):
+        facilities = FacilityIndex.objects.order_by("id").all()
+        size_of_chunk = 100000
+        count = facilities.count()
+        print_with_time(f"Amount of indexed facilities: {count}")
+
+        filename = create_with_headers()
+
+        index = 0
+        print_with_time("Fetching data from DB")
+        while count > index:
+            if index == 0:
+                print_with_time(
+                    f"Started exporting {count} facilities"
+                )
+
+            facility_chunk = create_chunk(facilities, size_of_chunk, index)
```

Comment:
What if you just write them all down at once?

Reply:
@vladsha-dev Yes, I got all facilities but then append to the file by chunks, so by each chunk it open & modify csv file.
It's improve perfomance more than twice, because we didn't do one operation during all facilities, we create operation by each chunk.

Reply:
@vladsha-dev The issue with memory & perfomance was when we modify it all down at once.

Reply:
Previously it trying to use more than 16GB of memory and now it less than 8-10GB.